File: IKeep.Lib/Services/Dtos/TreeNode.cs
﻿using IKeep.Lib.Models;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;


namespace IKeep.Lib.Services.Dtos
{
    public class TreeNode 
    {

        public Guid Id;

        public Guid ParentId;

        [JsonIgnore]
        public TreeNode Parent;
        public string Type;

        public string Name;

        public List<TreeNode> Children = new List<TreeNode>();
    }
}

File: IKeep.Web/ClientApp/src/services/dtos/TreeNode.js
class TreeNode
{
    constructor(json)
    {
        if(json)
        {
            this.Id = json.id,
            this.ParentId = json.parentId,
            this.New = json.new ? json.new : false,
            this.Type = json.type
        }
        else
        {
            this.Id = null,
            this.ParentId = "",
            this.New = false,
            this.Type = ""
        }
    }
}

export default TreeNode